lib/executor.js

```diff
@@ -274,7 +274,10 @@ function setupMiddleware(app, instructions) {
         data.fragment ? ('#' + data.fragment) : '');
     var factory = require(data.sourceFile);
     if (data.fragment) {
-      factory = factory[data.fragment];
+      if (typeof factory == 'function')
+        factory = factory[data.fragment];
+      else
+        factory = factory[data.fragment].bind(factory);
```

Comment:
What is the point of the first if-branch? Why can't we bind the factory method always?